Disable StressTestDeepNestingOfLoops for NonBacktracking engine (#87369)
dotnet/runtime: docs/workflow/testing/libraries/testing.md

dotnet build /t:Test
```

> This one command builds the library under test, then the test library, then runs the tests in the test library.

Reviewer: This isn't generally the case.  It's true for S.C.Immutable because the test project has a project reference to the src project, but for example the System.Net.Http test project does not have a project reference to the src project and does not cause the library under test to be built.

> This one command builds the library under test, then the test library, then runs the tests in the test library.

Author: ```suggestion
```

- `-nonamespace`
- `-parallel`

### Viewing XUnit logs

> It's usually sufficient to see the test failure output in the console. There is a test log file though and you can find it in a location like `...\runtime\artifacts\bin\System.Text.RegularExpressions.Tests\Debug\net8.0\testResults.xml`. It can be helpful for example to grep through a series of failures, or see how long a slow test actually took.

Reviewer: ```suggestion
It's usually sufficient to see the test failure output in the console. There is also a test log file, which you can find in a location like `...\runtime\artifacts\bin\System.Text.RegularExpressions.Tests\Debug\net8.0\testResults.xml`. It can be helpful, for example, to grep through a series of failures, or to see how long a slow test actually took.
```
